components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -1462,193 +1193,198 @@ public void testROIAndRenderingSettingsNoSudo(boolean permWriteOwned, boolean pe
             link = (ImageAnnotationLink) iQuery.findByQuery("FROM ImageAnnotationLink l JOIN FETCH"
                     + " l.child JOIN FETCH l.parent WHERE l.child.id = :id",
                     new ParametersI().addId(fileAnnotation.getId()));
             assertOwnedBy(link, normalUser);
             assertOwnedBy(fileAnnotation, normalUser);
-        } else {/* link was created but could not be chowned */
+            assertOwnedBy(originalFile, normalUser);
+        } else {
+            /* Second case: link was created but could not be chowned.*/
             link = (ImageAnnotationLink) iQuery.findByQuery("FROM ImageAnnotationLink l JOIN FETCH"
                     + " l.child JOIN FETCH l.parent WHERE l.child.id = :id",
                     new ParametersI().addId(fileAnnotation.getId()));
             assertOwnedBy(link, lightAdmin);
         }
     }
 
-    /** Test of light admin without using Sudo.
-     * The workflow tries to upload an official script.
-     * The only permission light admin needs for this is WriteScriptRepo
+    /**
+     * Light admin (lightAdming) tries to upload an official script.
+     * lightAdmin succeeds in this if they have <tt>WriteScriptRepo<tt> permission.
      * @param isPrivileged if to test a user who has the <tt>WriteScriptRepo</tt> privilege
      * @param groupPermissions if to test the effect of group permission level
      * @throws Exception unexpected
      */
     @Test(dataProvider = "isPrivileged cases")
     public void testOfficialSciptUploadNoSudo(boolean isPrivileged, String groupPermissions) throws Exception {
-        /* upload/creation of File Attachment should be always permitted as long as light admin is in System Group
-         * and has WriteOwned and WriteFile permissions. */
+        /* isPrivileged translates in this test into WriteScriptRepo permission, see below.*/
         boolean isExpectSuccessUploadOfficialScript = isPrivileged;
         final EventContext normalUser = newUserAndGroup(groupPermissions);
-        /* set up the light admin's permissions for this test */
+        /* Set up the light admin's permissions for this test.*/
         List<String> permissions = new ArrayList<String>();
         if (isPrivileged) permissions.add(AdminPrivilegeWriteScriptRepo.value);
         final EventContext lightAdmin;
         lightAdmin = loginNewAdmin(true, permissions);
         client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
         IScriptPrx iScript = factory.getScriptService();
-        /* fetch a script from the server */
+        /* lightAdmin fetches a script from the server.*/
         OriginalFile scriptFile = iScript.getScriptsByMimetype(ScriptServiceTest.PYTHON_MIMETYPE).get(0);
         RawFileStorePrx rfs = factory.createRawFileStore();
         rfs.setFileId(scriptFile.getId().getValue());
         final String actualScript = new String(rfs.read(0, (int) rfs.size()), StandardCharsets.UTF_8);
         rfs.close();
```

Comment:
try/finally since I am not sure we can use try-with-resource here (double-check) is required otherwise the ``RawFileStore`` might never be closed